### PR TITLE
Refactor package structure and remove unused method

### PR DESCRIPTION
**Pull request by eemihauk**
Connected to #149

screenshot from 2015-02-18 15 05 22

There are some files in the utils package that could be perhaps in their own packages. Let me know your opinions.

### CODE REVIEW

File: App/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -32,7 +32,7 @@
         android:label="@string/app_name"
         android:theme="@style/FSecureTheme">
         <activity
-            android:name="cc.softwarefactory.lokki.android.MainActivity"
+            android:name=".activities.MainActivity"
```
Comment (Kauhsa) [inline App/src/main/AndroidManifest.xml]: This is some kind of shorthand, apparently?

**Comment by Kauhsa** [comment]
Other than what you already said – "some files in the utils package that could be perhaps in their own packages" – it looks perfect. At least ContactUtils classes could be in their own package, since there are two of them.

**Comment by Kauhsa** [comment]
Actually, I'm not sure "utils" is a very good name for those classes. They're something that would be called "services" in Spring (yes, I know we can't use that here)... I wonder if there's something like that in Android terminology. "source"?

**Comment by eemihauk** [comment]
Might "helpers" be better for the "utils" package name? Also I don't understand what you mean by "source"...

**Comment by Kauhsa** [comment]
Yeah, sorry. The package name is fine. I'm talking about *ContactUtils classes specifically, because ContactUtils sounds like a silly interface name when a lot more specific name can be given: it defines a source where contact information can be got from.

**Comment by Kauhsa** [comment]
To clarify, the "Utils" name bothers me when used in interfaces for DI – interface defined for DI purposes (or well, any interface) shouldn't be just a collection of random crap, which is what "Utils" name signifies. For any other random crap, it's perfectly fine.

**Comment by eemihauk** [comment]
How about `ContactsProviderHelper` as it is used with Androids [Contacts Provider](http://developer.android.com/guide/topics/providers/contacts-provider.html)?

**Comment by Kauhsa** [comment]
Helper means just as much "random crap" as Utils does. Besides, the interface doesn't have anything to do with Android Contacts Provider.

**Comment by eemihauk** [comment]
Oh I see your edit to your earlier comment, that makes sense. Now I see what you mean and source is a good enough suffix for now at least I think. So I'll change the name to `ContactsSource`. I think I will rename the method `listContacts` method of the interface to be more descriptive of what the method needs to do.

**Comment by Kauhsa** [comment]
I was already thinking about asking @josalmi to explain this to you live because I'm clearly unable to express myself in textual format ;)

**Comment by eemihauk** [comment]
Current structure:
screenshot from 2015-02-18 17 38 27

**Comment by Kauhsa** [comment]
Looks good to me. I see you renamed some other classes as well to follow Java naming conventions, good ;) I glanced over the code changes as well and it looked OK.

**Comment by vforteli** [comment]
btw, the unused methods.. sure they arent these callbacks made using name of the method?

**Comment by josalmi** [comment]
Maybe ContactDataSource

**Comment by eemihauk** [comment]
Yes I made sure it was not found anywhere else by searching for the string.

**Comment by Kauhsa** [comment]
@josalmi I'm not sure "data" adds anything to the name but I'm not against

**Comment by Kauhsa** [comment]
Looks otherwise ok to me now, but I think the *Source classes don't need to be under `utils` package necessarily. Maybe have a `sources` package similar to `fragments`, `activities` etc? I can envision more of them coming in the future.

**Comment by eemihauk** [comment]
Current package structure of the PR:

screenshot from 2015-02-20 10 14 50

**Comment by Kauhsa** [comment]
Maybe scrap the underscore from `data_sources` since I haven't seen that ever in package names

**Comment by josalmi** [comment]
+1 for ^^

**Comment by eemihauk** [comment]
Renamed `data_sources` to `datasources`

**Pull request merged**